bee-block/src/output/alias.rs

```diff
@@ -437,27 +437,21 @@ impl AliasOutput {
         };
         let next_state = context.output_chains.get(&ChainId::from(alias_id));
 
-        let locked_address = self.unlock_conditions().locked_address(
-            match next_state {
-                Some(Output::Alias(next_state)) => {
-                    if self.state_index() == next_state.state_index() {
-                        self.governor_address()
-                    } else {
-                        self.state_controller_address()
-                    }
+        match next_state {
+            Some(Output::Alias(next_state)) => {
+                if self.state_index() == next_state.state_index() {
+                    self.governor_address().unlock(unlock, inputs, context)?;
+                } else {
+                    self.state_controller_address().unlock(unlock, inputs, context)?;
+                    context
```

Comment:
Maybe good to add a comment here that the alias is only unlocked when it's a state transition, otherwise it could look like we forgot it for the state controller when not reading the tip

Reply:
Sure thing